### ISSUE TITLE
Error handler doesn't get triggered when using non-async routes

### ISSUE THREAD

**Issue by yamalight** [fastify/fastify#1514]
## 🐛 Bug Report

`throw`ing an error from non-async route handler crashes the whole app.

## To Reproduce

```js
const app = require('fastify')();

app.get('/throw', (req, reply) => {
  throw new Error('I crash the app');
});
app.get('/throwAsync', async (req, reply) => {
  throw new Error('I work');
});
app.setErrorHandler((error, req, reply) => {
  reply.send(`Internal Server Error - ${error}`);
});

app.listen(8080, () => console.log(`Started at 8080`));
```

Opening `/throwAsync` will correctly show the error handler.
Opening `/throw` will crash the app.

## Expected behavior

I've expected throwing is `async` and normal functions to work the same way - i.e. passing the error to `setErrorHandler` function and not crashing the app.

## Your Environment

- *node version*: 11.10.1
- *fastify version*: >=2.0.1
- *os*: Mac

**Comment by jsumners** [comment]
https://github.com/fastify/fastify/issues/1078
https://github.com/fastify/fastify/issues/850

**Comment by yamalight** [comment]
@jsumners that makes sense. I feel like this should be better explained in the docs.
Currently [it says](https://github.com/fastify/fastify/blob/master/docs/Errors.md):

> Note that, while Fastify doesn't catch uncaught errors for you, if routes are declared as async, the error will safely be caught by the promise and routed to the default error handler of Fastify for a generic Internal Server Error response. For customizing this behaviour, you should use setErrorHandler.

I don't think that's clear enough - at least for me it wasn't and I assumed errors in sync routes are handled in the same fashion 🤔 
Changing it to something like this would make it more apparent:

> Note that, while Fastify doesn't catch uncaught errors for you, so any uncaught errors will result in crash.
> If routes are declared as async though - the error will safely be caught by the promise and routed to the default error handler of Fastify for a generic Internal Server Error response. For customizing this behaviour, you should use setErrorHandler.

Thoughts?

**Comment by jsumners** [comment]
> Note that Fastify doesn't catch uncaught errors within callback based routes for you, so any uncaught errors will result in crash.
If routes are declared as async though - the error will safely be caught by the promise and routed to the default error handler of Fastify for a generic Internal Server Error response. For customizing this behaviour, you should use setErrorHandler.

^ PR with that if you're able.

**Comment by yamalight** [comment]
@jsumners will send in a minute! 👍

**Comment by delvedor** [comment]
Closed in #1515.
